istioctl/pkg/auth/upgrader.go

```diff
@@ -0,0 +1,376 @@
+	sourceIP             = "source.ip"
+	requestAuthPrincipal = "request.auth.principal"
+)
+
+func NewUpgrader(k8sClient *kubernetes.Clientset, v1PolicyFiles, serviceFiles []string) *Upgrader {
+	upgrader := Upgrader{
+		K8sClient:                          k8sClient,
+		V1PolicyFiles:                      v1PolicyFiles,
+		ServiceFiles:                       serviceFiles,
+		NamespaceToServiceToWorkloadLabels: make(map[string]ServiceToWorkloadLabels),
+		AuthorizationPolicies:              []model.Config{},
+	}
+	return &upgrader
+}
+
+// ConvertV1alpha1ToV1beta1 converts RBAC v1alphal1 to v1beta1 for local policy files.
+func (ug *Upgrader) ConvertV1alpha1ToV1beta1() error {
+	configsFromFile, err := getRbacConfigsFromFiles(ug.V1PolicyFiles)
+	if err != nil {
+		return fmt.Errorf("failed to get RBAC config from files with error %v", err)
+	}
+	authzPolicies, err := getAuthorizationPolicies(configsFromFile)
+	if err != nil {
+		return fmt.Errorf("failed to get AuthorizationPolices store with error %v", err)
+	}
+	err = ug.convert(authzPolicies)
+	if err != nil {
+		return fmt.Errorf("failed to convert policies with error %v", err)
+	}
+	for _, authzPolicy := range ug.AuthorizationPolicies {
+		err := ug.parseConfigToString(authzPolicy)
+		if err != nil {
+			return fmt.Errorf("failed to parse config to string with error %v", err)
+		}
+	}
+	return nil
+}
+
+// getRbacConfigsFromFiles parses RBAC policy files and store them in an internal data type.
+func getRbacConfigsFromFiles(fileNames []string) ([]model.Config, error) {
+	configsFromFile := []model.Config{}
+	for _, fileName := range fileNames {
+		rbacFileBuf, err := ioutil.ReadFile(fileName)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read file %s", fileName)
+		}
+		configFromFile, _, err := crd.ParseInputs(string(rbacFileBuf))
+		if err != nil {
+			return nil, err
+		}
+		configsFromFile = append(configsFromFile, configFromFile...)
+	}
+	return configsFromFile, nil
+}
+
+// getAuthorizationPolicies creates an Istio store for the provided policies and returns all RBAC policy
+// in form of AuthorizationPolicies struct.
+func getAuthorizationPolicies(policies []model.Config) (*model.AuthorizationPolicies, error) {
+	store := model.MakeIstioStore(memory.Make(schemas.Istio))
+	for _, p := range policies {
+		if _, err := store.Create(p); err != nil {
+			return nil, fmt.Errorf("failed to initialize authz policies: %s", err)
+		}
+	}
+	authzPolicies, err := model.GetAuthorizationPolicies(&model.Environment{
+		IstioConfigStore: store,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create authz policies: %s", err)
+	}
+	return authzPolicies, nil
+}
+
+// convert is the main function that converts RBAC v1alphal1 to v1beta1 for local policy files
+func (ug *Upgrader) convert(authzPolicies *model.AuthorizationPolicies) error {
+	namespaces := authzPolicies.ListNamespacesOfToV1alpha1Policies()
+	if len(namespaces) == 0 {
+		return fmt.Errorf("no namespace found")
```

Comment:
Make this message longer so the user who gets it knows the problem is with the input, and not with the cluster.